### PR TITLE
Add mhv correlation ID to IAM user/identity

### PR DESCRIPTION
**Pull request by patrickvinograd**
## Description of change
Adds  mhv correlation ID parsed  from IAM OAuth token into IAM user/useridentity models, to  enable API access to MHV upstream data. 

A couple of TODOs but this should enable  testing of secure messaging API access for mobile. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

### CODE REVIEW

File: app/models/iam_user_identity.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ def self.build_from_iam_profile(iam_profile)
       iam_icn: iam_profile[:fediam_mviicn],
       iam_edipi: iam_profile[:fediam_do_dedipn_id],
       iam_sec_id: iam_profile[:fediamsecid],
+      iam_mhv_id: valid_mhv_id(iam_profile[:fediam_mhvien]),
```
Comment (jjulian) [inline app/models/iam_user_identity.rb:32]:
```suggestion
      iam_mhv_id: safe_mhv_id(iam_profile[:fediam_mhvien]),
```

**Comment by jjulian** [comment]
fyi rubocop not happy with your trailing whitespace

File: app/models/iam_user_identity.rb [hunk 2]
```diff
@@ -52,4 +53,14 @@ def set_expire
   def uuid
     Digest::UUID.uuid_v5(@iam_sec_id, @iam_icn)
   end
+
+  # Return a single mhv id from a possible comma-separated list value attribute
+  def self.valid_mhv_id(id_from_profile)
+    # TODO: site login fails if multiple _different_ MHV IDs are present
+    # Should this code be doing the same? Or at least logging a warning?
```
Comment (kreek) [inline app/models/iam_user_identity.rb:60]: Some questions about this.

-  What kind of issues does this cause, would their messages potentially diverge between accounts? 
- How does a veteran resolve having multiple MHV ids?

If the issues doesn't cause data issues, I think logging would be sufficient.

**Review by kreek — APPROVED** [review]
One question, but overall LGTM

File: app/models/iam_user_identity.rb (hunk 2 above)
Comment (patrickvinograd) [reply app/models/iam_user_identity.rb:61]: Yes, they'd potentially have different message contents between the two accounts.

MHV website  and VA.gov (as of the SSOe integration) prevent a user in such state from signing  in,  and presents a  message instructing them to call the help desk to resolve.

Comment (kreek) [reply app/models/iam_user_identity.rb:61]: 👍 ok, sounds like we should do the same thing as .gov and raise an error (and map that error to a helpful in-app message)

**Comment by patrickvinograd** [comment]
This still needs specs, and has gotten a little unwieldy. My plan is to  split it out into 3  PRs:
1) Adding the MHV ID to IAMUserIdentity
2) Adding the module-specific SM client  and configuration
3) All the controllers/serializers.

**Comment by patrickvinograd** [comment]
Obsoleted by subsequent PRs

**Pull request closed without merging**